core/domain/exp_jobs_one_off.py

```diff
@@ -128,19 +142,29 @@ def entity_classes_to_map_over(cls):
 
     @staticmethod
     def map(item):
+        is_valid_math_expression = schema_utils.get_validator(
+            'is_valid_math_expression')
+        is_valid_math_equation = schema_utils.get_validator(
+            'is_valid_math_equation')
+        ltt = latex2text.LatexNodes2Text()
+
         if not item.deleted:
             exploration = exp_fetchers.get_exploration_from_model(item)
             for state_name, state in exploration.states.items():
                 if state.interaction.id == 'MathExpressionInput':
                     for group in state.interaction.answer_groups:
                         for rule_spec in group.rule_specs:
-                            rule = rule_spec.inputs['x']
-                            if any(sep in rule for sep in SEPARATORS):
-                                yield (
-                                    item.id,
-                                    '%s: %s' % (
-                                        state_name.encode('utf-8'),
-                                        rule.encode('utf-8')))
+                            rule_input = ltt.latex_to_text(
+                                rule_spec.inputs['x'])
+                            validity = 'Invalid'
+                            if is_valid_math_expression(rule_input):
+                                validity = 'Valid Expression'
+                            elif is_valid_math_equation(rule_input):
+                                validity = 'Valid Equation'
+                            yield (
+                                item.id,
+                                u'%s: %s -- %s' % (
+                                    state_name, rule_input, validity))
```

Comment:
This seems a bit off. If what you want to do is view the problematic cases rather than list all formulae, then perhaps your first element (the key for the mapreduce) should be by validity and the second element should contain all the details. For the valid cases you probably can just output the first N (100-1000?) rule_input strings for eyeballing, but for the invalid cases you'll want to output all of them with full details.

Reply:
That makes sense. I've set the limit for valid explorations to 200.